subgraphs/_reference_/src/common/getters.ts

```diff
@@ -0,0 +1,115 @@
+// import { log } from "@graphprotocol/graph-ts"
+import { Address, BigInt, ethereum } from "@graphprotocol/graph-ts";
+import {
+  _Account,
+  _DailyActiveAccount,
+  Token,
+  DexAmmProtocol,
+  LiquidityPool,
+  UsageMetricsDailySnapshot,
+  FinancialsDailySnapshot,
+  PoolDailySnapshot,
+} from "../../generated/schema";
+import { fetchTokenSymbol, fetchTokenName, fetchTokenDecimals } from "./tokens";
+import {
+  BIGDECIMAL_ZERO,
+  Network,
+  INT_ZERO,
+  FACTORY_ADDRESS,
+  ProtocolType,
+  SECONDS_PER_DAY,
+} from "../common/constants";
+
+export function getOrCreateToken(tokenAddress: Address): Token {
+  let token = Token.load(tokenAddress.toHexString());
+  // fetch info if null
+  if (!token) {
+    token = new Token(tokenAddress.toHexString());
+    token.symbol = fetchTokenSymbol(tokenAddress);
+    token.name = fetchTokenName(tokenAddress);
+    token.decimals = fetchTokenDecimals(tokenAddress);
+    token.save();
+  }
+  return token;
+}
+
+export function getOrCreateUsageMetricSnapshot(event: ethereum.Event): UsageMetricsDailySnapshot {
```

Comment:
Any chance common functions like this, live in a shared location instead, to get rid of copy-and-paste?

Reply:
Yeah, I've thought about that too. The problem is they depend on generated types from both the schema and the ABIs, which makes things a little bit more tricky

Reply:
Can we do this for utilities that depend only on schema?

Reply:
I'm not sure what's the best way to do that but I do think that's better in the long run. I don't have the capacity to work on this right now. Would you like to take a stab at it? If not, I can probably get to this towards the end of next week.

I'm going to close out this PR for now but feel free to start a new one.